homeassistant/components/solax/sensor.py

```diff
@@ -47,7 +47,7 @@
         key=f"{Units.KWH}_{False}",
         device_class=SensorDeviceClass.ENERGY,
         native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
-        state_class=SensorStateClass.MEASUREMENT,
+        state_class=SensorStateClass.TOTAL_INCREASING,
```

Comment:
This change feels incorrect, as the definition below does provide this.

It seems to be tied into the `is_monitic` flag. What else is using this?

Reply:
Those 2 entries are similar, but distinct. The non-monotonic one, the one modified in this change request is the total energy generated for the day.

The next sensor, which is similar but marked as monotonic, is the total lifetime energy generation of the unit.

This issue is that the sensor is not providing a "last_reset" value to the dashboard.

Here you can see the values coming off the SolaX unit for last week.
![image](https://github.com/home-assistant/core/assets/2999162/8594996a-0674-48d6-8241-045e2a090d01)

This is the relevant error in the energy dashboard configuration menu:
![image](https://github.com/home-assistant/core/assets/2999162/409f54c4-cf28-4388-90e0-42f4096ee1d8)

According to the information that I could find, changing the sensor state class to total_increasing seems to be a reasonable fix, as the decreasing value is interpreted as the resetting of the period.

[Excerpt from here](https://developers.home-assistant.io/blog/2021/08/16/state_class_total/)

> For sensors with state_class STATE_CLASS_TOTAL_INCREASING, a decreasing value is interpreted as the start of a new meter cycle or the replacement of the meter. It is important that the integration ensures that the value cannot erroneously decrease in the case of calculating a value from a sensor with measurement noise present.

I'm new the the internals of Home Assistant, so I'm happy to take guidance on what a more correct approach may be to this problem, if this turns out to be the wrong one.

Reply:
Also attaching an image to show the difference in the monotonic and non-monotonic KWH values specified:
![image](https://github.com/home-assistant/core/assets/2999162/9259b0bc-8de8-4d9b-a247-4ff92fef190f)

Reply:
A week and a half ago I had a chat with codeowner @squishykid, we'll have another one at the end of Feb.

I have been working on stuff have formed an opinion on this: almost all KwH sensors should be of the monotonic variety and only the remaining battery capacity should be non-monotonic and switch device_class to ENERY_BATTERY. (Figuring out wether it should be `SensorStateClass.TOTAL`  or `SensorStateClass.MEASUREMENT` what I will be figuring out when I finish writing this comment.)

Which I think means I'll have to make sure the first update every calendar day should set `last_reset` for the ones that reset daily, probably just all. (Otherwise there will be edge cases due to the 10% tolerance.

TL;DR: It can't be fixed with a one liner on the HA side. Changes on both sides are needed.

Reply:
Should this not be state_class=SensorStateClass.TOTAL_INCREASING as stated in the change. The amount increases during the day and then is reset at 0:00 with it increasing at the next da. 
<img width="1036" alt="Screenshot 2024-03-29 at 13 11 29" src="https://github.com/home-assistant/core/assets/105568090/b104785d-d09f-4a84-96a8-9ed70f1145ef">

Reply:
`SensorStateClass.TOTAL` with last reset prevents/lessens the impact of an edge case.
If home-assistant/wifi goes down when the daily total is 2kWh and only comes back up the next day when it is already at 2.4kWh than it would only register an .4kWh increase with `SensorStateClass.TOTAL_INCREASING` while `SensorStateClass.TOTAL` will register 2.4kWh.